apps/web-portal-client/lib/features/home/hooks/usePreviews.jsx

```diff
@@ -0,0 +1,51 @@
+import {useEffect, useContext, useReducer} from 'react'
+import {AppContext} from '../../../context'
+
+export const usePreviews = () => {
+  const fetchPreviewsReducer = (state, action) => {
+    switch (action.type) {
+      case 'FETCH_INIT': {
+        return {...state, isLoading: true, isError: false}
+      }
+      case 'FETCH_SUCCESS': {
+        return {...state, isLoading: false, previews: action.payload}
+      }
+      case 'FETCH_FAILURE': {
+        return {...state, isLoading: false, isError: true}
+      }
+      default:
+        return state
+    }
+  }
+
+  const [state, dispatch] = useReducer(fetchPreviewsReducer, {
+    previews: undefined,
+    isLoading: false,
+    isError: false,
+  })
+
+  const {api} = useContext(AppContext)
+
+  useEffect(() => {
```

Comment:
I think we should remove that effect because usage of 'doFetch' from outside would do the job

Reply:
ok, but this way we should don't forget that we need to prevent updating our state when unmounting. I mean flag didCancel. https://www.robinwieruch.de/react-hooks-fetch-data

Reply:
Reviewed, good point let's include it in next task for extracting common effect for reuse